src/App.js

```diff
@@ -1,14 +1,134 @@
 import React from 'react';
+import TodoList from './components/TodoComponents/TodoList';
+import TodoForm  from './components/TodoComponents/TodoForm';
+
 
 class App extends React.Component {
-  // you will need a place to store your state in this component.
-  // design `App` to be the parent component of your application.
-  // this component is going to take care of state, and any change handlers you need to work with your state
-  render() {
-    return (
-      <div>
-        <h2>Welcome to your Todo App!</h2>
-      </div>
+    state = {
+        items: [],
+        task: '',
+        error: null,
+        search: '',
+    }
+    
+    addItem = event => {
+        event.preventDefault();
+        if(this.state.task === ''){
+            return this.setState({
+                error: 'You must enter a task!'
+            })
+        }
+        const newItem = {
+            task: this.state.task,
+            id: Date.now(),
+            completed: false,
+            matching: true
+        }
+
+        const tempList = [...this.state.items, newItem];
+        const sortedList = tempList.sort((a,b) => {
+            const taskA = a.task;
+            const taskB = b.task;
+            if(taskA > taskB) return 1
+            else return -1
+        }); 
+
+        this.setState(state => ({
+            items: sortedList,
+            task: '',
+            error: null
+        }))
+
+        
+    }
+    
+    handleChanges = event => {
+        const test = event.target.name;
+        this.setState({
+            [event.target.name]: event.target.value
+        },() => {
+            if(test === 'search'){
+                this.searchItems();
+            }
+        })
+    }
+
+    completeTask = (id) => {
+        const completedItem = this.state.items.filter(item => item.id === id)[0]
+        completedItem.completed = !completedItem.completed;
+        this.setState(prevState => ({
```

Comment:
does this work? It doesn't really look like you are making any changes to the array. Might confuse some people trying to read your code.